sklearn/metrics/_regression.py

```diff
@@ -387,7 +394,8 @@ def median_absolute_error(y_true, y_pred, *, multioutput='uniform_average'):
     """
     y_type, y_true, y_pred, multioutput = _check_reg_targets(
         y_true, y_pred, multioutput)
-    output_errors = np.median(np.abs(y_pred - y_true), axis=0)
+    output_errors = _weighted_percentile(np.abs(y_pred - y_true),
+                                         sample_weight=sample_weight)
```

Comment:
I think we should avoid calling `_weighted_percentile` if sample_weight is None and still rely on `np.median`. The reason being that `np.median` is probably much faster than our more general `_weighted_percentile` (could be wrong on that, but I'd be surprised).